File: src/main/java/net/kigawa/kweb/entity/Service.java
package net.kigawa.kweb.entity;

import javax.persistence.*;

@Entity
public class Service
{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false)
    private Long id;
    @Column(nullable = false, unique = true)
    private String strId;
    @Column
    private String topImg;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Column
    private String title;

    public Long getId()
    {
        return id;
    }

    public void setId(Long id)
    {
        this.id = id;
    }

    public String getStrId()
    {
        return strId;
    }

    public void setStrId(String strId)
    {
        this.strId = strId;
    }

    public String getTopImg()
    {
        return topImg;
    }

    public void setTopImg(String topUrl)
    {
        this.topImg = topUrl;
    }
}

File: src/main/java/net/kigawa/kweb/entity/Service.kt
package net.kigawa.kweb.entity

import java.net.URL
import javax.persistence.*

@Entity
open class Service protected constructor() {
    companion object {
        fun create(topImg: URL, title: String, strId: String, subtitle: String): Service {
            val service = Service()
            service.strId = strId
            service.topImg = topImg
            service.title = title
            service.subtitle = subtitle
            return service
        }
    }

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false)
    open var id: Long = 0

    @Column(nullable = false, unique = true)
    open var strId: String = ""

    @Column
    open lateinit var topImg: URL

    @Column
    open var title: String = ""

    @Column
    open var subtitle: String = ""
}